lib/runners/processor/clang_tidy.rb

```diff
@@ -0,0 +1,99 @@
+module Runners
+  class Processor::ClangTidy < Processor
+    Schema = StrongJSON.new do
+      let :runner_config, Schema::BaseConfig.base.update_fields { |fields|
+        fields.merge!({
+          apt: enum?(string, array(string)),
+          'include-path': enum?(string, array(string))
+        })
+      }
+
+      let :issue, object(
+        severity: string,
+      )
+    end
+
+    register_config_schema(name: :clang_tidy, schema: Schema.runner_config)
+
+    # @see https://github.com/github/linguist/blob/775b07d40c04ef6e0051a541886a8f1e30a892f4/lib/linguist/languages.yml#L532-L535
+    # @see https://github.com/github/linguist/blob/775b07d40c04ef6e0051a541886a8f1e30a892f4/lib/linguist/languages.yml#L568-L584
+    VALID_EXTENSIONS = [".c", ".cpp", ".c++", ".cc", ".cp", ".cxx"].freeze
+    GLOB_HEADERS = "**/*.{h,h++,hh,hpp,hxx,inc,inl,ipp,tcc,tpp}".freeze
+
+    def setup
+      deploy_packages
+      yield
+    end
+
+    def analyzer_bin
+      "clang-tidy"
+    end
+
+    def analyze(changes)
+      issues = []
+
+      changes
+        .changed_paths
+        .select { |path| VALID_EXTENSIONS.include?(path.extname.downcase) }
+        .map{ |path| relative_path(working_dir / path, from: current_dir) }
+        .reject { |path| path.to_s.start_with?("../") } # reject files outside the current_dir
+        .each do |path|
+          stdout, = capture3!(analyzer_bin, path.to_s, "--", *option_include_path,
+            is_success: ->(status) { [0, 1].include?(status.exitstatus) })
+          construct_result(stdout) { |issue| issues << issue }
+        end
+
+      Results::Success.new(guid: guid, analyzer: analyzer).tap do |result|
+        result.add_issue(*issues)
+      end
+    end
+
+    private
+
+    def deploy_packages
+      # select "lib*" and report others as warning for security concerns
+      packages = Array(config_linter[:apt])
+        .select do |pkg|
+          if pkg.start_with?("lib")
+            true
+          else
+            add_warning "Installing the package `#{pkg}` is blocked."
```

Comment:
[suggestion] The warning is due to `sider.yml`, so can you add `file`, please?

```suggestion
            add_warning "Installing the package `#{pkg}` is blocked.", file: config.path_name
```

Reply:
LGTM.
commit f0fa9e6c31248e842659e94fad41774d04d63a50